FBSnapshotTestCase/Categories/UIImage+Compare.m

```diff
@@ -129,4 +115,63 @@ - (BOOL)fb_compareWithImage:(UIImage *)image tolerance:(CGFloat)tolerance
     return imageEqual;
 }
 
+- (BOOL)_comparePixelWithPixelTolerance:(CGFloat)pixelTolerance
+                         referencePixel:(FBComparePixel *)referencePixel
+                             imagePixel:(FBComparePixel *)imagePixel
+{
+    if (referencePixel->raw == imagePixel->raw) {
+        return YES;
+    } else if (pixelTolerance == 0) {
+        return NO;
+    }
+
+    CGFloat redPercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.red
+                                                                     imagePixelComponent:imagePixel->pixels.red];
+    CGFloat greenPercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.green
+                                                                       imagePixelComponent:imagePixel->pixels.green];
+    CGFloat bluePercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.blue
+                                                                      imagePixelComponent:imagePixel->pixels.blue];
+    CGFloat alphaPercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.alpha
+                                                                       imagePixelComponent:imagePixel->pixels.alpha];
+
+    return !(redPercentDiff > pixelTolerance ||
+             greenPercentDiff > pixelTolerance ||
+             bluePercentDiff > pixelTolerance ||
+             alphaPercentDiff > pixelTolerance);
+}
+
+- (CGFloat)_calculatePercentDifferenceForReferencePixelComponent:(char)p1
+                                             imagePixelComponent:(char)p2
+{
+    int referencePixelComponent = (unsigned char)p1;
+    int imagePixelComponent = (unsigned char)p2;
+    int componentDifference = abs(referencePixelComponent - imagePixelComponent);
+    return (CGFloat)componentDifference / 256;
+}
+
+- (BOOL)_comparePixelWithPixelTolerance:(CGFloat)pixelTolerance
+                              tolerance:(CGFloat)tolerance
+                             pixelCount:(NSInteger)pixelCount
+                        referencePixels:(FBComparePixel *)referencePixel
+                            imagePixels:(FBComparePixel *)imagePixel
+{
+    NSInteger numDiffPixels = 0;
```

Comment:
nit: my convention is to always use `NSUInteger` when I know negative values aren't needed, if you have time we should update these where used.

Reply:
Done 👍